### DISCUSSION TITLE
open file from :Git diff

### DISCUSSION THREAD

**Discussion by onsails** [tpope/vim-fugitive#1734] [Q&A] [answered]
I often look `:Git diff`  for presence of debug lines which should be deleted after commit.
Is there any way to open line under cursor in an editable buffer?

**Comment by tpope** [comment, marked as answer]
[`<CR>`](https://github.com/tpope/vim-fugitive/blob/895e56daca03c441427f2adca291cb10ea4d7ca8/doc/fugitive.txt#L310-L316). Note that if you can't tell just by looking what exactly is on each side off the diff, Fugitive ain't gonna be able to tell either. `diff.mnemonicPrefix` can improve things a bit.

**Reply by chen965** [reply]
222

**Reply by vitaly-zdanevich** [reply]
Return (Enter) opens a buffer with a content from .git :(

**Reply by Hritik14** [reply]
@vitaly-zdanevich exactly what I was looking for, thanks!

**Reply by tpope** [reply]
> Return (Enter) opens a buffer with a content from .git :(

It only does this if the surrounding context makes clear you're looking at an old diff, and this is a good thing because the line numbers might not line up otherwise, or the code might even be deleted entirely. You can navigate from that old buffer to the work tree (automatically adjusting the line numbers if possible) with `:Gedit`.
